File: src/main/kotlin/com/huawei/cangjie1/psi/CjModifierListOwner.java
package com.huawei.cangjie1.psi;

import com.huawei.cangjie1.lexer.CjModifierKeywordToken;
import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;


public interface CjModifierListOwner extends PsiElement ,CjElement {
    @Nullable
    CjModifierList getModifierList();

    boolean hasModifier(@NotNull CjModifierKeywordToken modifier);

    void addModifier(@NotNull CjModifierKeywordToken modifier);
    void removeModifier(@NotNull CjModifierKeywordToken modifier);


}

File: src/main/kotlin/com/linqingying/cangjie/psi/CjModifierListOwner.kt
package com.linqingying.cangjie.psi

import com.linqingying.cangjie.descriptors.DescriptorVisibility
import com.intellij.psi.PsiElement
import com.linqingying.cangjie.lexer.CjKeywordToken

interface CjModifierListOwner : PsiElement, CjAnnotated {

    val modifierList: CjModifierList?

    fun hasModifier(modifier: CjKeywordToken): Boolean

    fun addModifier(modifier: CjKeywordToken)

    fun removeModifier(modifier: CjKeywordToken)

    val modifierVisibility: DescriptorVisibility?
}
